### PR TITLE
chore: add well npt filtering [release]

### PR DESCRIPTION
**Pull request by Sollimann**
[SDL-649] - all/any filter on multiple npt codes and npt code details

[SDL-649]: https://cognitedata.atlassian.net/browse/SDL-649

### CODE REVIEW

File: packages/wells/README.md [hunk 1]
```diff
@@ -164,15 +164,55 @@ const retrievedCrs = wells.items.map(well => well.wellhead?.crs)
...
 
+This allows you to filter wells on associated npt events.
+
+**duration**: how long did the event last.
+
+**measuredDepth**: at what depth the event occured.
+
+**nptCodes**: what npt codes are associated to the event.
+
+**nptCodeDetail**: what nptCodeDetails are associated to the event
```
Comment (sighol) [inline packages/wells/README.md]: super nit: `npt codes` on line 173 and `nptCodeDetails` on line 175. In the description.

File: packages/wells/README.md [hunk 2]
```diff
@@ -164,15 +164,55 @@ const retrievedCrs = wells.items.map(well => well.wellhead?.crs)
 
 ### _Filter wells by npt events_
 
+This allows you to filter wells on associated npt events.
+
+**duration**: how long did the event last.
```
Comment (sighol) [inline packages/wells/README.md]: Duration in *seconds*.

File: packages/wells/README.md [hunk 3]
```diff
@@ -393,15 +433,21 @@ const data: SurveyData = await client.surveys.getData(request);
 
 #### _Filter NPT events_
 
+duration: how long did the event last
```
Comment (sighol) [inline packages/wells/README.md]: duration in seconds.

File: packages/wells/src/client/model/WellNPTFilter.ts [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+export interface WellNPTFilter {
+  /**
+   * @type {LengthRange}
+   * @memberof WellNPTFilter
+   */
+  measuredDepth?: LengthRange;
+  /**
+   * @type {DoubleRange}
+   * @memberof WellNPTFilter
+   */
```
Comment (sighol) [inline packages/wells/src/client/model/WellNPTFilter.ts:13]: Curious: What does these comments add? Both the type and the `memberof` property can be deduced from the code. Does it improve IDE integration? Doesn't typescript extract this info from the code already?

**Review by sighol — APPROVED** [review]
The duration is in _seconds_, right? I think we should state the unit in the documentation.

File: packages/wells/README.md (hunk 2 above)
Comment (Sollimann) [reply packages/wells/README.md]: duration is in hours. I'll specify that

File: packages/wells/README.md (hunk 3 above)
Comment (Sollimann) [reply packages/wells/README.md]: it's in hours

File: packages/wells/src/client/model/WellNPTFilter.ts (hunk 4 above)
Comment (Sollimann) [reply packages/wells/src/client/model/WellNPTFilter.ts:13]: I copied the pattern for the geospatial nodejs sdk. I haven't tested it properly yet and whether or not it works with or without, but it's mainly intended for the autogenerated docs. If you do `yarn docs` inside of the repo and open the `index.html` in a live server, you'll get our autogenerated docs. 
image

If you go into any of the exported classes, you'll find a reference to memberof. 
image

We should release this like the rest of Cognite at some point. Javier could in fact use this documentation to share with Academy.

File: packages/wells/README.md (hunk 2 above)
Comment (Sollimann) [reply packages/wells/README.md]: In the source data this is already stored in hours

File: packages/wells/src/client/model/WellNPTFilter.ts (hunk 4 above)
Comment (Sollimann) [reply packages/wells/src/client/model/WellNPTFilter.ts:13]: I'll create a card for releasing it.

Comment (Sollimann) [reply packages/wells/src/client/model/WellNPTFilter.ts:13]: created [[SDL-665]](https://cognitedata.atlassian.net/secure/RapidBoard.jspa?rapidView=391&projectKey=SDL&modal=detail&selectedIssue=SDL-665&assignee=5d1c705543d1510d3accfb4a)

**Pull request merged**